core/modules/system/system.install

```diff
@@ -110,6 +110,19 @@ function system_requirements($phase) {
     $requirements['php_register_globals']['value'] = $t('Disabled');
   }
 
+  // Test PHP max_input_vars setting.
+  $max_input_vars = ini_get('max_input_vars');
+  $requirements['php_max_input_vars'] = array(
+    'title' => $t('PHP max_input_vars setting'),
+    'value' => $t('PHP limits the number of input variables in each form to @value items.', array('@value' => number_format($max_input_vars, 0, '.', ','))),
+  );
+  // 1000 is the default value and indicates potential issues.
+  // @see https://www.php.net/manual/en/info.configuration.php
+  if ($max_input_vars <= 1000) {
+    $requirements['php_max_input_vars']['description'] = $t('When set too low, the <code>max_input_vars</code> PHP setting can cause silent failures on large forms, such as the permissions page. A value of <code>10000</code> is recommended. <a href="@documentation" target="_blank">Read more about increasing the <code>max_input_vars</code> PHP setting</a>.', array('@documentation' => url('https://docs.backdropcms.org/max-input-vars')));
+    $requirements['php_max_input_vars']['severity'] = REQUIREMENT_WARNING;
```

Comment:
Perhaps downgrade the notice so this does not throw a warning on installation and updates:

```suggestion
    $requirements['php_max_input_vars']['severity'] = $phase === 'runtime' ? REQUIREMENT_WARNING : REQUIREMENT_INFO;
```

Like @indigoxela said in the issue, that might be causing the failures. I think update.php puts in an interstitial warning page if you have active warnings during update.

Reply:
there shouldn't be any active warnings if the server is set up correctly?

Reply:
Update: @quicksketch says that other updates also have a special status during install (and update) that are different from the actual status that appears on the status report. I'm going to rebase and first, and go from there :)

Reply:
okay, I added 
```php
    if ($phase == 'install' || $phase == 'update') {
      $requirements['php_max_input_vars']['severity'] = REQUIREMENT_INFO;
    }
```